content/blog/2020/release-0.2.0.md

```diff
@@ -0,0 +1,190 @@
+of a user if the change is later spent together with other outputs.
+
+### Key generation
+
+If you need to generate a new `bip32::ExtendedPrivKey`, or perhaps a new BIP39 mnemonic, you can use the unified [`GeneratableKey`][`GeneratableKey`] trait to do so: paired with [`GeneratableDefaultOptions`][`GeneratableDefaultOptions`] they provide many different ways to generate keys,
+with or without a custom source of entropy, and with or without customized options.
+
+```rust
+use bdk::bitcoin::PrivateKey;
+use bdk::keys::{GeneratableKey, GeneratableDefaultOptions, PrivateKeyGenerateOptions};
+
+let default_options_key = PrivateKey::generate_default()?;
+let custom_options_key = PrivateKey::generate(PrivateKeyGenerateOptions { compressed: false })?;
+```
+
+### Generic key types
+
+With this update there's now a generalized trait for keys that can be used in descriptors, which is called [`ToDescriptorKey`][`ToDescriptorKey`]. This trait is already implemented for the native `rust-bitcoin` key types, like `PrivateKey`, `PublicKey`, `bip32::ExtendedPrivKey`
+and `bip32::ExtendedPubKey`. It's also implemented for BIP39 mnemonic and seeds, when the the opt-in `keys-bip39` feature is enabled. As always, being this a public trait, you can also implement it for custom types to better suit your needs.
+
+```rust
+impl<Ctx: ScriptContext> ToDescriptorKey<Ctx> for MyKeyType {
+    fn to_descriptor_key(self) -> Result<DescriptorKey<Ctx>, KeyError> {
+        // Custom conversion to `bitcoin::PrivateKey`
+        let privkey: bitcoin::PrivateKey = ... ;
+        privkey.to_descriptor_key()
+    }
+}
+```
+
+If your custom key types is simply a different representation of an `xprv` or `xpub`, you can also consider implementing the [`DerivableKey`][`DerivableKey`] trait instead: for a type `K` that implements [`DerivableKey`][`DerivableKey`], the [`ToDescriptorKey`][`ToDescriptorKey`] trait is automatically
+implemented for the [`(K, bip32::DerivationPath)`][K_path] and [`(K, bip32::KeySource, bip32::DerivationPath)`][K_src_path] tuples.
+
+```rust
+impl<Ctx: ScriptContext> DerivableKey<Ctx> for MyKeyType {
+     fn add_metadata(
+        self, 
+        origin: Option<KeySource>, 
+        derivation_path: DerivationPath
+    ) -> Result<DescriptorKey<Ctx>, KeyError> {
+        // Custom conversion to `bip32::ExtendedPrivKey`
+        let xprv: bip32::ExtendedPrivKey = ... ;
+        xprv.add_metadata(origin, derivation_path)
+    }
+}
+```
+
+### Descriptor templates
+
+Instead of having to serialize keys to strings using `format!()` just to place them somewhere inside a descriptor, you can now use descriptor templates to build a descriptor starting from a key and some other options
+in a couple of lines of code. You can use one of the [provided templates][desc_templates_mod] or make a custom one by implementing the [`DescriptorTemplate`][`DescriptorTemplate`] trait on a `struct` or `enum`.
```

Comment:
This link isn't showing up when rendered: [provided templates][desc_templates_mod]

Reply:
There were backticks around the link name, so it was seeing it as a different link. Should be fixed now

Reply:
Looks good now